resources/dynamodb_tables.go

```diff
@@ -0,0 +1,714 @@
+						Name:        "replica_status",
+						Description: "The current state of the replica:  * CREATING - The replica is being created.  * UPDATING - The replica is being updated.  * DELETING - The replica is being deleted.  * ACTIVE - The replica is ready for use.",
+						Type:        schema.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+func fetchDynamodbTables(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	c := meta.(*client.Client)
+	svc := c.Services().DynamoDB
+
+	config := dynamodb.ListTablesInput{}
+	for {
+		output, err := svc.ListTables(ctx, &config, func(o *dynamodb.Options) {
+			o.Region = c.Region
+		})
+		if err != nil {
+			return err
+		}
+
+		for i := range output.TableNames {
+			response, err := svc.DescribeTable(ctx, &dynamodb.DescribeTableInput{TableName: &output.TableNames[i]}, func(o *dynamodb.Options) {
+				o.Region = c.Region
+			})
+			if err != nil {
+				return err
+			}
+			res <- response.Table
+		}
+
+		if aws.ToString(output.LastEvaluatedTableName) == "" {
+			break
+		}
+		config.ExclusiveStartTableName = output.LastEvaluatedTableName
+	}
+
+	return nil
+}
+func resolveDynamodbTableTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	table, ok := resource.Item.(*types.TableDescription)
+	if !ok {
+		return fmt.Errorf("expected *types.TableDescription but got %T", resource.Item)
+	}
+
+	cl := meta.(*client.Client)
+	svc := cl.Services().DynamoDB
+	response, err := svc.ListTagsOfResource(ctx, &dynamodb.ListTagsOfResourceInput{
+		ResourceArn: table.TableArn,
+	}, func(options *dynamodb.Options) {
+		options.Region = cl.Region
+	})
+	if err != nil {
+		return err
+	}
+
+	tags := make(map[string]interface{})
+	for _, t := range response.Tags {
+		tags[*t.Key] = t.Value
+	}
+	return resource.Set(c.Name, tags)
+}
+func resolveDynamodbTableArchivalSummary(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(*types.TableDescription)
+	if r.ArchivalSummary == nil {
+		return nil
+	}
+	return resource.Set(c.Name, map[string]interface{}{
+		"date_time":  r.ArchivalSummary.ArchivalDateTime,
+		"backup_arn": r.ArchivalSummary.ArchivalBackupArn,
+		"reason":     r.ArchivalSummary.ArchivalReason,
+	})
+}
+func resolveDynamodbTableAttributeDefinitions(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(*types.TableDescription)
+	val := make([]map[string]interface{}, len(r.AttributeDefinitions))
+	for i := range r.AttributeDefinitions {
+		val[i] = map[string]interface{}{
```

Comment:
Question: Keep this or make it a map of Name vs. Object(type,name)? Order shouldn't matter anyway.

Reply:
I would simply encode `AttributeDefinitions` as JSON. But it's fine as is too. Map would allow easier retrieval of type by name.